src/main/java/com/google/devtools/build/lib/packages/StarlarkSemanticsOptions.java

```diff
@@ -287,6 +287,19 @@
               + " https://github.com/bazelbuild/bazel/issues/8830 for details.")
   public boolean experimentalAllowTagsPropagation;
 
+  @Option(
+      name = "experimental_allow_recursion",
```

Comment:
This should not be a flag, because we don't want to expose it to the Bazel command line. It should be an API option set explicitly by clients that need it. So you can revert this file.